### DISCUSSION TITLE
Running without drone

### DISCUSSION THREAD

**Discussion by luislglp** [jyjblrd/Mocap-Drones#49] [Q&A] [answered]
I was trying to run the code without a drone, so I simply commented the lines:
ser = serial.Serial("/dev/cu.usbserial-02X2K2GE", 1000000, write_timeout=1, )
cameras.set_ser(ser)

As long as I do not use the drone functions on the front end, I would guess it should work fine. 

I am running on Windows 11 (followed the steps in https://github.com/bensondaled/pseyepy/issues/15).
The pseyepy library is working fine (tested with the basic usage example in https://github.com/bensondaled/pseyepy). I also changed the code to not rely on the java front end and it seems to be working (at least the capture points part). Notice that I am running with 2 cameras, without the IR mod yet =)
![image2](https://github.com/jyjblrd/Low-Cost-Mocap/assets/127667739/4c457cdb-384e-4cbb-ab89-62a3eaa4c18f)

But it seems that there is no communication between the backend and the front end. Whenever I start the camera stream, nothing happens. 
![image1](https://github.com/jyjblrd/Low-Cost-Mocap/assets/127667739/e063fcc6-4915-4838-a37e-0348cf4b6a91)

Not sure if I am missing a basic step here, maybe firewall/administrator settings?

**Comment by jyjblrd** [comment]
It should work, try using the chrome network tools to see if it is connecting to the backend, and maybe add some print statements to the backend to see if requests are being received from the frontend. You may need to reload the page after resetting the python script to see the images show up.

**Reply by luislglp** [reply]
Thanks for the hints @jyjblrd 

So I entered in the chrome network tools (ctrl+shit+i shortcut) and there were indeed some issues with comm: "Failed to load resource: net::ERR_CONNECTION_REFUSED".

I noticed that whenever I was running the backend I got the following message:
"WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.  * Running on http://127.0.0.1:5000". Maybe the index.py does not use the port 3001 as in "socketio.run(app, port=3001, debug=True)". Not sure why, but it actually overrides with the standard 5000 option. 

My workaround was to change in "App.tsx" the local host 3001 to 5000. In the following line of code: 

In291: const fetchResponse = await fetch(`http://localhost:3001/api/trajectory-planning`, settings);

ln377: <img src={cameraStreamRunning ? "http://localhost:3001/api/camera-stream" : ""} />

After that I started receiving camera stream. But I am not familiar with Java and web dev, not sure if this is the right correction since it seems that I am still receiving some connection issues in the front end (see image).

![image](https://github.com/jyjblrd/Low-Cost-Mocap/assets/127667739/8c15caf4-00f0-4652-b663-edd4e2e7ff76)

**Reply by jyjblrd** [reply, marked as answer]
There are a few more places where you will need to change the socket port if you choose to resolve the problem this way: https://github.com/search?q=repo%3Ajyjblrd%2FLow-Cost-Mocap%203001&type=code
